grails-app/controllers/org/pih/warehouse/api/OrganizationApiController.groovy

```diff
@@ -41,9 +41,7 @@ class OrganizationApiController extends BaseDomainApiController {
         Organization organization = new Organization()
         bindOrganizationData(organization, jsonObject)
 
-        if (organization.hasErrors() || !organization.save(flush: true)) {
-            throw new ValidationException("Invalid organization", organization.errors)
-        }
+        organizationService.saveOrganization(organization)
```

Comment:
Let's see if we can remove the first three line 
```
JSONObject jsonObject = request.JSON
Organization organization = new Organization()
bindOrganizationData(organization, jsonObject)
```
and replace the method with this 
```
def create(Organization organization) { 
    organizationService.saveOrganization(organization)
    render([data: [id: organization.id]] as JSON)
}
```
If that works in the happy case, does it actually need to be 
```
def create(Organization organization) { 
    organization = organizationService.saveOrganization(organization)
    render([data: [id: organization.id]] as JSON)
}
```
or something completely different?

Reply:
`bindOrganizationData` generates `organization.code` and `organization.partyType`
```
        if (!organization.code) {
            organization.code =
                    identifierService.generateOrganizationIdentifier(organization.name)
        }

        if (!organization.partyType) {
            organization.partyType = PartyType.findByCode(Constants.DEFAULT_ORGANIZATION_CODE)
        }
```
so it's necessary to be working properly (without this bind validation doesn't pass).

Reply:
@alannadolny I think this part could be moved out of the bind into create action with the Organization param, I think this will look cleaner. Other than that lgtm.

Reply:
Yeah, this is borderline "business logic". I would either move it into the controller like @awalkowiak said OR add a createOrganization() method in the service to handle this logic. Create would call saveOrganization().